### PR TITLE
Add keyboard shortcuts to modals

### PR DESCRIPTION
**Pull request by KeeJef**
Working off this PR https://github.com/oxen-io/session-desktop/pull/2281 which adds shortcuts for the 'Delete' and 'Backspace' 
keys to delete a message when a user is in selection mode. 

This PR extends additional keyboard shortcuts for modals created using the SessionConfirm dialog component, adding 'Escape' which will call the onClickCancelHandler to close the modal and 'Enter' which will call the onClickOkHandler. This should make navigating deletion dialogs easier.

### CODE REVIEW

File: ts/components/conversation/SessionConversation.tsx
Comment (Bilb) [inline ts/components/conversation/SessionConversation.tsx:61]: is this needed at all otherwise lgtm

**Review by Bilb — CHANGES_REQUESTED** [review]

File: ts/components/conversation/SessionConversation.tsx
Comment (Bilb) [reply ts/components/conversation/SessionConversation.tsx:61]: conflicts need to be resolved with unstable

Comment (KeeJef) [reply ts/components/conversation/SessionConversation.tsx]: Rebased and resolved comment

**Review by Bilb — APPROVED** [review]

**Comment by yougotwill** [comment]
Hi @KeeJef I have some concerns about this PR. Can we discuss further before merging?

File: ts/components/conversation/SessionConversation.tsx
Comment (yougotwill) [inline ts/components/conversation/SessionConversation.tsx:359]: You need to account for messages that you can only delete locally. For example being a non-admin in a group or a community.

```tsx
            if (selectedConversationKey) {
              if (isOnlyServerDeletable) {
                void onDeleteSelectedMessagesForEveryone(
                  selectedConversationKey,
                  selectedMessageIds
                );
              } else {
                void deleteMessagesById(selectedMessageIds, selectedConversationKey);
              }
            }
  ```

Comment (yougotwill) [reply ts/components/conversation/SessionConversation.tsx:359]: const isOnlyServerDeletable = isPublic;

**Review by yougotwill — CHANGES_REQUESTED** [review]
Looking good! one small change required

File: ts/components/conversation/composition/CompositionButtons.tsx
Comment (yougotwill) [inline ts/components/conversation/composition/CompositionButtons.tsx:53]: if we get to React 19 one day we won't need forwardRef anymore as ref is included in all components 🙏

File: ts/components/conversation/header/ConversationHeaderSelectionOverlay.tsx
Comment (yougotwill) [inline ts/components/conversation/header/ConversationHeaderSelectionOverlay.tsx:38]: why do we need the useKey here is we listen for key down on the Session Convo? What am I missing?

File: ts/components/icon/SessionIconButton.tsx
Comment (yougotwill) [inline ts/components/icon/SessionIconButton.tsx:22]: Are we 100% sure that changing to button doesn't change any existing styles or layout anywhere? For example I can't remember if a button is a block or inline element but a div is a block element.

**Review by yougotwill — CHANGES_REQUESTED** [review]
Nice to see more accessibility happening!

File: ts/components/conversation/composition/CompositionButtons.tsx
Comment (Bilb) [reply ts/components/conversation/composition/CompositionButtons.tsx:53]: well we will right, as part of onboarding?

File: ts/components/conversation/header/ConversationHeaderSelectionOverlay.tsx
Comment (Bilb) [reply ts/components/conversation/header/ConversationHeaderSelectionOverlay.tsx:38]: because as soon as you select a message, the focus is token by the selection overlay so those actions are not doing anything.

File: ts/components/basic/SessionButton.tsx
Comment (Bilb) [reply ts/components/basic/SessionButton.tsx:71]: should be the right color, but I think the animation is making the start color white, which transitions to the primary-color. hence maybe the weird color at the start?

File: ts/components/SessionWrapperModal.tsx
Comment (Bilb) [reply ts/components/SessionWrapperModal.tsx:74]: I am not sure I follow, the button on line 114 does not mount unless if onConfirm is truthy right?
```ts
 {onConfirm ? (
                    <SessionButton buttonType={SessionButtonType.Simple} onClick={props.onConfirm}>
                      {confirmText || window.i18n('ok')}
                    </SessionButton>
                  ) : null}

Comment (Bilb) [reply ts/components/SessionWrapperModal.tsx:74]: seems like this work role='button'

File: ts/components/icon/SessionIconButton.tsx
Comment (Bilb) [reply ts/components/icon/SessionIconButton.tsx:22]: I am pretty sure it's ok, I had a look already. If really, it shouldn't break anything, just have a style issue until we fix it. And QA might find some

File: ts/components/basic/SessionButton.tsx
Comment (yougotwill) [reply ts/components/basic/SessionButton.tsx]: That is strange. Do we want the animation? Otherwise we can remove the transition?

File: ts/components/conversation/composition/CompositionButtons.tsx
Comment (yougotwill) [reply ts/components/conversation/composition/CompositionButtons.tsx:53]: Sadly 19 isn't out yet. We do have 18 in onboarding though 👀

File: ts/components/conversation/header/ConversationHeaderSelectionOverlay.tsx
Comment (yougotwill) [reply ts/components/conversation/header/ConversationHeaderSelectionOverlay.tsx:38]: Okay. Might be worth leaving a comment to say that is the plan in future?

File: ts/components/icon/SessionIconButton.tsx
Comment (yougotwill) [reply ts/components/icon/SessionIconButton.tsx:22]: Cool. Thanks for checking

File: ts/components/basic/SessionRadio.tsx
Comment (yougotwill) [inline ts/components/basic/SessionRadio.tsx:90]: Very nice! 🧙

File: ts/components/basic/SessionButton.tsx
Comment (Bilb) [reply ts/components/basic/SessionButton.tsx]: well we don't need one in the end, this outline is now removed (and we use the browser default)

Comment (yougotwill) [reply ts/components/basic/SessionButton.tsx]: Cool cool

**Review by yougotwill — APPROVED** [review]
LGTM! 🚢

File: ts/components/conversation/header/ConversationHeaderSelectionOverlay.tsx
Comment (yougotwill) [inline ts/components/conversation/header/ConversationHeaderSelectionOverlay.tsx:78]: I'm thinking we should use the convention of `fallbackFocusId`?

**Review by yougotwill — CHANGES_REQUESTED** [review]

File: ts/components/conversation/header/ConversationHeaderSelectionOverlay.tsx
Comment (Bilb) [reply ts/components/conversation/header/ConversationHeaderSelectionOverlay.tsx:78]: well we can't have the same id twice in the document right. If we used the same one for the SelectionOverlay and the dialog, when they are both displayed it's going to be messy

Comment (yougotwill) [reply ts/components/conversation/header/ConversationHeaderSelectionOverlay.tsx:78]: All good. I misread that classNameAndID is used for the styles and id as well

**Review by yougotwill — APPROVED** [review]
LGTM! 🚢

**Pull request merged**
